src/main/java/org/openstreetmap/atlas/geography/atlas/delta/AtlasDeltaGenerator.java

```diff
@@ -55,44 +71,118 @@ protected SwitchList switches()
         return new SwitchList().with(BEFORE_SWITCH, AFTER_SWITCH, OUTPUT_DIRECTORY_SWITCH);
     }
 
-    private void compare(final Atlas beforeAtlas, final Atlas afterAtlas, final Path outputDir)
+    private void run(final Path before, final Path after, final Path outputDirectory)
     {
-        final String name = beforeAtlas.getName().split(".atlas")[0];
+        final Time time = Time.now();
+
+        this.logger.info("Comparing {} and {}", before, after);
+
+        // If the after is a directory, we want to diff the individual shards in parallel.
+        if (Files.isDirectory(after))
+        {
+            // You need to have the before dir be a dir of shards too for this to work.
+            if (!Files.isDirectory(before))
+            {
+                logger.error("Your -before parameter must point to a directory of atlas shards if "
+                        + "you want to compare shard by shard with an -after directory also of shards!");
+                System.exit(64);
+            }
+
+            // Execute in a pool of threads so we limit how many atlases get loaded in parallel.
+            final ForkJoinPool customThreadPool = new ForkJoinPool(THREADS);
+            try
+            {
+                customThreadPool.submit(() -> this.compareShardByShard(before, after, outputDirectory))
+                        .get();
+            }
+            catch (final InterruptedException interrupt)
+            {
+                logger.error("The shard diff workers were interrupted.", interrupt);
+            }
+            catch (final ExecutionException exec)
+            {
+                logger.error("There was an execution exception on the shard diff workers.", exec);
+            }
+        }
+        // Otherwise, we can do a normal compare where we look at 2 atlases or input shards with a
+        // single output.
+        else
+        {
+            final Atlas beforeAtlas = load(before);
+            final Atlas afterAtlas = load(after);
+            compare(beforeAtlas, afterAtlas, outputDirectory);
+        }
+
+        logger.info("AtlasDeltaGenerator complete. Total time: {}.", time.elapsedSince());
+    }
+
+    /**
+     * Load a multi atlas if directory, otherwise load single atlas.
+     *
+     * @param path
+     *            An atlas shard directory or a single atlas.
+     * @return An atlas object.
+     * @throws IOException
+     *             Exception if loading the atlas directory does not work.
+     */
+    private Atlas load(final Path path)
+    {
+        return Files.isDirectory(path) ? loadAtlasDirectory(path) : loadSingleAtlas(path);
```

Comment:
nit: "this." before local function calls.

Reply:
Right, will do.